src/lib.rs

```diff
@@ -280,60 +314,120 @@ impl Log for SimpleLogger {
+                                Level::Debug => record.level().to_string().purple().to_string(),
+                                Level::Trace => record.level().to_string().normal().to_string(),
+                            },
+                            LVL_FMT,
+                        )
                     } else {
-                        record.level().to_string()
+                        Field::new(record.level().to_string(), LVL_FMT)
                     }
                 }
                 #[cfg(not(feature = "colored"))]
                 {
-                    record.level().to_string()
+                    Field::new(record.level().to_string(), LVL_FMT)
                 }
             };
 
             let target = if !record.target().is_empty() {
-                record.target()
+                Field::new(record.target(), TGT_FMT)
             } else {
-                record.module_path().unwrap_or_default()
+                Field::new(record.module_path().unwrap_or_default(), TGT_FMT)
             };
 
+            #[cfg(feature = "thread_ids")]
+            let thread_id = std::thread::current();
+
+            #[cfg(feature = "thread_ids")]
+            let target_thread = if self.threadids {
+                Field::new(thread_id.name().unwrap_or("UNKNOWN"), TRD_FMT)
+            } else {
+                Field::none()
+            };
+
+            #[cfg(not(feature = "thread_ids"))]
+            let target_thread = "";
+
             #[cfg(feature = "chrono")]
-            if self.timestamps {
-                #[cfg(not(feature = "stderr"))]
-                println!(
-                    "{} {:<5} [{}] {}",
-                    Local::now().format("%Y-%m-%d %H:%M:%S,%3f"),
-                    level_string,
-                    target,
-                    record.args()
-                );
-                #[cfg(feature = "stderr")]
-                eprintln!(
-                    "{} {:<5} [{}] {}",
-                    Local::now().format("%Y-%m-%d %H:%M:%S,%3f"),
-                    level_string,
-                    target,
-                    record.args()
-                );
-                return;
-            }
+            let timestamp = if self.timestamps {
+                Field::new(Local::now().format("%Y-%m-%d %H:%M:%S,%3f"), TS_FMT)
+            } else {
+                Field::none()
+            };
+
+            #[cfg(not(feature = "chrono"))]
+            let timestamp = Field::<char>::none();
 
             #[cfg(not(feature = "stderr"))]
-            println!("{:<5} [{}] {}", level_string, target, record.args());
+            println!(
+                LOGGING_FMT!(),
+                ts = timestamp,
+                lvl = level_string,
+                thr = target_thread,
+                tgt = target,
+                msg = record.args()
+            );
+
             #[cfg(feature = "stderr")]
-            eprintln!("{:<5} [{}] {}", level_string, target, record.args());
```

Comment:
We've lost the `{:<5}` formatting for the level string with this change. I'm also not sure how spacing between the fields is done?

Reply:
turns out this is a showstopper, i am working on this now. In case i don't find a solution (just passing {lvl:<5} isn't passed through, i have to rtfm) this may make the whole thing impossible.

Reply:
It can go directly in `LOGGING_FMT` rather than being done in `fn log`, I think.

Reply:
unfortunally not, because the `LOGGING_FMT "{something}"   becomes a `"{something_prefix}{something}{something_suffix}"` in Field, there the alignment gets lost. I'll look into that, may take a few days, there is no urge here.

Reply:
That makes sense. Alternatively `level` doesn't need to use `Field` like the others - it's always present, always used in the format and doesn't have any suffix/prefix.

Reply:
yes that would be some 'last resort' that works. But want to see if it can be made generic first. Other fields my benefit from such formatting rules eventually.